[VAULT-35715] UI: Namespace picker is updated after add/delete namespace

beagins opened the pull request:
### Description
What does this PR do?
- [x] Namespace picker list is updated after add/delete namespace on the manage namespace page w/o manual refresh
- [ ] Enterprise tests passing


https://github.com/user-attachments/assets/15e306c2-fad0-476d-8efc-8c732eff9fac

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

beagins commented on ui/tests/acceptance/enterprise-namespaces-test.js:
```diff
@@ -15,63 +15,17 @@ import {
-    await settled();
-  }
-}
-
```
renamed and moved to `helpers/commands.js`

hellobontempo commented on ui/tests/acceptance/access/namespaces/index-test.js:
Perhaps I'm misunderstanding the PR title/description - but it looks like we still need a test to assert that the namespace list updates when delete/add happens _without_ clicking "Refresh"?

beagins replied:
Thank you for catching this Claire! I've added the following test to cover this use case: 
`it should update namespace list after create/delete without manual refresh`

beagins commented:
>277 tests completed in 159396 milliseconds, with 0 failed, 10 skipped, and 0 todo.
1607 assertions of 1607 passed, 0 failed.

Monkeychip commented on ui/app/components/namespace-picker.hbs:
```diff
@@ -6,7 +6,12 @@
+      @icon="org"
+      @text={{or this.selectedNamespace.id "-"}}
+      @isFullWidth={{true}}
+      data-test-namespace-toggle
```
There is a dynamic generic selector called data-test-toggle-group or data-test-toggle-input, would you mind using that syntax here? I should probably make this consist throughout.

beagins commented:
Enterprise tests are passing after addressing comments, rebasing from `main` and addressing merge conflicts!! 😅 ✅ 

>277 tests completed in 161084 milliseconds, with 0 failed, 10 skipped, and 0 todo. 
1602 assertions of 1602 passed, 0 failed.
